x/stakeibc/types/message_toggle_trade_controller_test.go

```diff
@@ -0,0 +1,94 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/Stride-Labs/stride/v19/app/apptesting"
+	"github.com/Stride-Labs/stride/v19/x/stakeibc/types"
+)
+
+func TestMsgToggleTradeController(t *testing.T) {
+	apptesting.SetupConfig()
+
+	validNotAdminAddress, invalidAddress := apptesting.GenerateTestAddrs()
+	validAdminAddress, ok := apptesting.GetAdminAddress()
+	require.True(t, ok)
+
+	validAddress := "cosmosXXX"
+	validChainId := "chain-0"
+	validPermissionChange := types.AuthzPermissionChange_GRANT
+
+	tests := []struct {
+		name string
+		msg  types.MsgToggleTradeController
+		err  string
+	}{
+		{
+			name: "valid message",
+			msg: types.MsgToggleTradeController{
+				Creator:          validAdminAddress,
+				ChainId:          validChainId,
+				PermissionChange: validPermissionChange,
+				Address:          validAddress,
+			},
+		},
+		{
+			name: "invalid address",
+			msg: types.MsgToggleTradeController{
+				Creator:          invalidAddress,
+				ChainId:          validChainId,
+				PermissionChange: validPermissionChange,
+				Address:          validAddress,
+			},
+			err: "invalid creator address",
+		},
+		{
+			name: "not admin address",
+			msg: types.MsgToggleTradeController{
+				Creator:          validNotAdminAddress,
+				ChainId:          validChainId,
+				PermissionChange: validPermissionChange,
+				Address:          validAddress,
+			},
+			err: "not an admin",
+		},
+		{
+			name: "invalid address",
+			msg: types.MsgToggleTradeController{
+				Creator:          validAdminAddress,
+				ChainId:          validChainId,
+				PermissionChange: validPermissionChange,
+				Address:          "",
+			},
+			err: "address must be specified",
+		},
+		{
+			name: "invalid chain ID",
+			msg: types.MsgToggleTradeController{
+				Creator:          validAdminAddress,
+				ChainId:          "",
+				PermissionChange: validPermissionChange,
```

Comment:
is it possible to test invalidPermissionChange too, or not because that's handled at the tx level?

Reply:
i think it is - just added a case for it